routers/repo/pull.go

```diff
@@ -810,6 +810,33 @@ func MergePullRequest(ctx *context.Context, form auth.MergePullRequestForm) {
 		return
 	}
 
+	lastCommitStatus, err := pull_service.GetPullRequestCommitStatusState(pr)
```

Comment:
if the pr's base branch is not protected. this will be error ...
```
D:/Go/src/runtime/panic.go:212 (0x44e843)
		panicmem: panic(memoryError)
	D:/Go/src/runtime/signal_windows.go:246 (0x44e6f0)
		sigpanic: panicmem()
	C:/gitea_work/gitea/services/pull/commit_status.go:129 (0x1617bd0)
		GetPullRequestCommitStatusState: return MergeRequiredContextsCommitStatus(commitStatuses, pr.ProtectedBranch.StatusCheckContexts), nil
	C:/gitea_work/gitea/routers/repo/pull.go:813 (0x19e7e3d)
		MergePullRequest: lastCommitStatus, err := pull_service.GetPullRequestCommitStatusState(pr)
	D:/Go/src/reflect/value.go:460 (0x49aea1)
```

Reply:
Fixed.